lib/topic_query.rb

```diff
@@ -995,7 +995,7 @@ def self.remove_muted_tags(list, user, opts = {})
           regular: CategoryUser.notification_levels[:regular],
           watching_or_infinite:
             (
-              if SiteSetting.watched_precedence_over_muted
+              if user&.watched_precedence_over_muted || SiteSetting.watched_precedence_over_muted
```

Comment:
Are we calling this code path for an anonymous user? I don't think that should be the case.

Reply:
Query itself is checking if user exists:

https://github.com/discourse/discourse/pull/22340/files#diff-288e61b8bacdb29d9c2e05b42da6837b0036dcf1867332d977ca7c5e74a44297L993

So this parameter is passed to query but not doing anything. I can try to add this parameter conditionally, but it may look a bit ugly. What do you think?

Reply:
I would just leave out the where clause related to notification levels for an anonymous user. That is just unnecessary work and makes the query more complicated than  it needs to be for an anonymous user. Do you know if this is a fairly straight forward change?